Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift

```diff
@@ -391,6 +391,8 @@ open class ChatChannelViewModel: ObservableObject, MessagesDataSource {
         if !isActive {
             return
         }
+        // Message data can change with every callback
+        utils.messageCachingUtils.clearCache()
```

Comment:
Not sure about this one. Another option could be to listen to specific events via an events controller?

Reply:
Thinking about closing this PR since the ticket about improving message list updates reverses these changes here. I think it is going to be better if I tackle this bug after those changes.